bundles/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/ConfigStore.java

```diff
@@ -0,0 +1,358 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.core.ConfigurableService;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.eclipse.smarthome.core.items.Item;
+import org.eclipse.smarthome.core.items.Metadata;
+import org.eclipse.smarthome.core.items.MetadataKey;
+import org.eclipse.smarthome.core.items.MetadataRegistry;
+import org.eclipse.smarthome.core.net.CidrAddress;
+import org.eclipse.smarthome.core.net.NetUtil;
+import org.eclipse.smarthome.core.net.NetworkAddressService;
+import org.openhab.io.hueemulation.internal.dto.HueAuthorizedConfig;
+import org.openhab.io.hueemulation.internal.dto.HueDataStore;
+import org.openhab.io.hueemulation.internal.dto.HueGroupEntry;
+import org.openhab.io.hueemulation.internal.dto.HueLightEntry;
+import org.openhab.io.hueemulation.internal.dto.HueRuleEntry;
+import org.openhab.io.hueemulation.internal.dto.HueSensorEntry;
+import org.openhab.io.hueemulation.internal.dto.response.HueSuccessGeneric;
+import org.openhab.io.hueemulation.internal.dto.response.HueSuccessResponseStateChanged;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * This component sets up the hue data store and gets the service configuration.
+ * It also determines the address for the upnp service by the given configuration.
+ * <p>
+ * Also manages the pairing timeout. The service is restarted after a pairing timeout, due to the ConfigAdmin
+ * configuration change.
+ * <p>
+ * This is a central component and required by all other components and may not
+ * depend on anything in this bundle.
+ *
+ * @author David Graeff - Initial contribution
+ */
+@Component(immediate = false, service = { ConfigStore.class }, configurationPid = {
+        HueEmulationService.CONFIG_PID }, property = { "com.eclipsesource.jaxrs.publish=false",
+                ConfigurableService.SERVICE_PROPERTY_DESCRIPTION_URI + "=io:hueemulation",
+                ConfigurableService.SERVICE_PROPERTY_CATEGORY + "=io",
+                ConfigurableService.SERVICE_PROPERTY_LABEL + "=Hue Emulation" })
+@NonNullByDefault
+public class ConfigStore {
+
+    public static final String METAKEY = "HUEEMU";
+    public static final String EVENT_ADDRESS_CHANGED = "ESH_EMU_CONFIG_ADDR_CHANGED";
+
+    private final Logger logger = LoggerFactory.getLogger(ConfigStore.class);
+
+    public HueDataStore ds = new HueDataStore();
```

Comment:
can you use a more expressive name here? like `dataStore`?

Reply:
The datastore is a central piece of the entire bundle and consistently called "ds". I'd rather not rename that to be honest.